packages/react/src/components/linkbox/Linkbox.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useRef } from 'react';
+
+// import core base styles
+import 'hds-core';
+import styles from './Linkbox.module.scss';
+import { IconArrowRight, IconLinkExternal } from '../../icons';
+import classNames from '../../utils/classNames';
+
+export type LinkboxProps = Omit<React.ComponentPropsWithoutRef<'a'>, 'tabIndex'> & {
+  /**
+   * Aria-label of the linkbox component.
+   */
+  ariaLabel: string;
+  /**
+   * Boolean indicating for external link that takes user to an entirely new web site. Defaults to false.
+   */
+  external?: boolean;
+  /**
+   * Optional heading of the linkbox.
+   */
+  heading?: string;
+  /**
+   * Heading level for linkbox title. Defaults to 2.
+   */
+  headingAriaLevel?: number;
+  /**
+   * The image props passed to the linkbox image. Image variant requires prop withImg set to true.
+   */
+  imgProps?: React.ComponentPropsWithoutRef<'img'>;
+  /**
+   * Boolean for the variant with no background color. Changes paddings also. Defaults to false.
+   */
+  noBackground?: boolean;
+  /**
+   * Aria label for opening link in an external domain. Active when external prop is true.
+   */
+  openInExternalDomainAriaLabel?: string;
+  /**
+   * Boolean indicating whether the link will open in new tab or not. Defaults to false.
+   */
+  openInNewTab?: boolean;
+  /**
+   * Aria label for opening link in a new tab
+   */
+  openInNewTabAriaLabel?: string;
+  /**
+   * Optional text of the linkbox.
+   */
+  text?: string;
+  /**
+   * Boolean indicating whether the Linkbox contains a border or not. Defaults to false.
+   */
+  withBorder?: boolean;
+  /**
+   * Boolean indicating whether the Linkbox contains a hero image or not. Defaults to false.
+   */
+  withImg?: boolean;
+  /**
+   * Size variant for the linkbox when accompanied with an img. Defaults to 'small'
+   */
+  withImgSize?: 'small' | 'medium' | 'large';
+};
+
+export const Linkbox = ({
+  ariaLabel,
+  children,
+  className,
+  external = false,
+  heading,
+  headingAriaLevel = 2,
+  href,
+  imgProps,
+  noBackground = false,
+  openInExternalDomainAriaLabel,
+  openInNewTab = false,
+  openInNewTabAriaLabel,
+  text,
+  withBorder = false,
+  withImg = false,
+  withImgSize = 'small',
+  ...rest
+}: LinkboxProps) => {
+  const linkRef = useRef(null);
+
+  const comboseAriaLabel = () => {
```

Comment:
I also notice Link has a similar function with the same name?
Can these two be combined or could they have some shared defaults?

Reply:
- [ ] Rename: combose -> compose

Reply:
> I also noticed Link has a similar function with the same name.
> Can these two be combined, or could they have some shared defaults, etc?

They are similar but contain some differences, so can't be combined without modification. They could have some shared defaults. Recently I've been leaning more towards 'repeat until it hurts' philosophy 😅, since it often helps reading the code because of less indirection. Maybe if they are repeated 3 times or more in the codebase, it would be good idea to combine then? We could combine them in some internal TS file that both components then apply.

Reply:
I would at least fix the typos 😁